### DISCUSSION TITLE
Can we add maxLength and other configs in text field of controls add-on?

### DISCUSSION THREAD

**Discussion by neetutanwar** [storybookjs/storybook#15027] [Ideas]
Storybook is great! 

We have started using it for building components, we do have a requirement to restrict the number of characters in copy and headers fields. We are using the **Controls** add-on for changing the content of a component, but currently, there is no option to pass configs to the text component of the controls add-on just like we have for the Number component to handle the min and max value.

I was wondering if we can have some basic and useful configs for the text input field. e.g.

- **MaxLength** - Option to enable max length property in the text input field to controls the no characters a field can store.

     something like this: 
      ![image](https://user-images.githubusercontent.com/22315597/119437948-beb2ac00-bd3c-11eb-9ecd-e7c6237fd086.png)

- **Validations** - Ability to pass regex and validate text input against it. 

file reference: https://github.com/storybookjs/storybook/blob/next/lib/components/src/controls/Text.tsx

**Comment by joeslazaro** [comment]
It looks like `maxLength` support was added to text controls in 2022, and the related file was moved from the location listed in the original issue description. It is now located at https://github.com/storybookjs/storybook/blob/next/code/ui/blocks/src/controls/Text.tsx
